app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt

```diff
@@ -21,37 +25,206 @@ import dagger.Component
+      onView(withId(R.id.action_audio_player)).perform(click())
+      onView(withId(R.id.action_audio_player)).perform(click())
+      onView(withId(R.id.ivPlayPauseAudio)).check(matches(isDisplayed()))
+      onView(withText(context.getString(R.string.cellular_data_alert_dialog_title))).check(doesNotExist())
+    }
+    explorationDataController.stopPlayingExploration()
+  }
+
+  @Test
+  fun testAudioWithWifi_openRatioExploration_clickAudioIcon_checkAudioFragmentHasDefaultLanguageAndAutoPlays() {
+    getApplicationDependencies(RATIOS_EXPLORATION_ID_0)
+    networkConnectionUtil.setCurrentConnectionStatus(NetworkConnectionUtil.ConnectionStatus.LOCAL)
+    ActivityScenario.launch<ExplorationActivity>(createExplorationActivityIntent(RATIOS_EXPLORATION_ID_0)).use {
+      onView(withId(R.id.action_audio_player)).perform(click())
+      onView(withId(R.id.ivPlayPauseAudio)).check(matches(isDisplayed()))
+      onView(withText("en")).check(matches(isDisplayed()))
+      onView(withId(R.id.ivPlayPauseAudio)).check(matches(withContentDescription(context.getString(R.string.audio_pause_description))))
+    }
+    explorationDataController.stopPlayingExploration()
+  }
+
+  @Test
+  fun testAudioWithWifi_openFractionsExploration_changeLanguage_clickNext_checkLanguageIsHinglish() {
+    getApplicationDependencies(FRACTIONS_EXPLORATION_ID_0)
+    networkConnectionUtil.setCurrentConnectionStatus(NetworkConnectionUtil.ConnectionStatus.LOCAL)
+    ActivityScenario.launch<ExplorationActivity>(createExplorationActivityIntent(FRACTIONS_EXPLORATION_ID_0)).use {
+      onView(withId(R.id.continue_button)).perform(click())
+      onView(withId(R.id.action_audio_player)).perform(click())
+      onView(withText("EN")).perform(click())
+      onView(withText("Hinglish")).perform(click())
+      onView(withText(context.getString(R.string.audio_language_select_dialog_okay_button))).perform(click())
+      onView(withId(R.id.continue_button)).perform(click())
+      onView(withText("HI-EN")).check(matches(isDisplayed()))
+    }
+    explorationDataController.stopPlayingExploration()
+  }
+
+  @Test
+  fun testAudioWithWifi_openRatioExploration_continueToInteraction_clickAudioButton_submitAnswer_checkFeedbackAudioPlays() {
+    getApplicationDependencies(RATIOS_EXPLORATION_ID_0)
+    networkConnectionUtil.setCurrentConnectionStatus(NetworkConnectionUtil.ConnectionStatus.LOCAL)
+    ActivityScenario.launch<ExplorationActivity>(createExplorationActivityIntent(RATIOS_EXPLORATION_ID_0)).use {
+      for (x in 0..4) {
+        onView(withId(R.id.continue_button)).perform(click())
+      }
+      onView(withId(R.id.action_audio_player)).perform(click())
+      onView(withId(R.id.text_input_interaction_view)).perform(ViewActions.typeText("123"), closeSoftKeyboard())
+      onView(withId(R.id.interaction_button)).perform(click())
+      // Ensures that feedback audio starts playing
+      Thread.sleep(1000)
```

Comment:
This may result in a flaky test. Is there a more reliable wait that we can do?

Reply:
Is this actually downloading audio, then? If we're using the real media player, we should probably ``@Ignore`` this test & related until we can actually swap out with a fake implementation.

Reply:
Updated to ignore test.